securedrop/models.py

```diff
@@ -141,139 +195,154 @@ def public_key(self, value: str) -> None:
     size = Column(Integer, nullable=False)
     downloaded = Column(Boolean, default=False)
-    '''
+    """
     The checksum of the encrypted file on disk.
     Format: $hash_name:$hex_encoded_hash_value
     Example: sha256:05fa5efd7d1b608ac1fbdf19a61a5a439d05b05225e81faa63fdd188296b614a
-    '''
+    """
     checksum = Column(String(255))
 
     def __init__(self, source: Source, filename: str) -> None:
         self.source_id = source.id
         self.filename = filename
         self.uuid = str(uuid.uuid4())
-        self.size = os.stat(current_app.storage.path(source.filesystem_id,
-                                                     filename)).st_size
+        self.size = os.stat(
+            current_app.storage.path(source.filesystem_id, filename)
+        ).st_size
 
     def __repr__(self) -> str:
-        return '<Submission %r>' % (self.filename)
+        return "<Submission %r>" % (self.filename)
 
-    def to_json(self) -> 'Dict[str, Union[str, int, bool]]':
+    def to_json(self) -> "Dict[str, Union[str, int, bool]]":
         json_submission = {
-            'source_url': url_for('api.single_source',
-                                  source_uuid=self.source.uuid) if self.source else None,
-            'submission_url': url_for('api.single_submission',
-                                      source_uuid=self.source.uuid,
-                                      submission_uuid=self.uuid) if self.source else None,
-            'filename': self.filename,
-            'size': self.size,
-            'is_read': self.downloaded,
-            'uuid': self.uuid,
-            'download_url': url_for('api.download_submission',
-                                    source_uuid=self.source.uuid,
-                                    submission_uuid=self.uuid) if self.source else None,
+            "source_url": url_for("api.single_source", source_uuid=self.source.uuid)
+            if self.source
+            else None,
+            "submission_url": url_for(
+                "api.single_submission",
+                source_uuid=self.source.uuid,
+                submission_uuid=self.uuid,
+            )
+            if self.source
+            else None,
+            "filename": self.filename,
+            "size": self.size,
+            "is_read": self.downloaded,
+            "uuid": self.uuid,
+            "download_url": url_for(
+                "api.download_submission",
+                source_uuid=self.source.uuid,
+                submission_uuid=self.uuid,
+            )
+            if self.source
+            else None,
         }
         return json_submission
 
+    @property
+    def seen(self) -> bool:
+        """
+        If a submission has been downloaded or seen by any journalist, then the submssion is
+        considered seen.
+        """
+        seen_file = SeenFile.query.filter_by(file_id=self.id).all()
+        seen_message = SeenMessage.query.filter_by(message_id=self.id).all()
```

Comment:
Do we technically need `.all()` here (for both queries) or would `.one()` suffice?